docs/extensions/release_notes.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import annotations
+
+import itertools
+import re
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+from docutils import nodes
+from packaging.version import Version
+from sphinx.util.docutils import SphinxDirective
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable, Sequence
+    from typing import ClassVar
+
+    from myst_parser.mdit_to_docutils.base import DocutilsRenderer
+    from sphinx.application import Sphinx
+
+
+FULL_VERSION_RE = re.compile(r"^(\d+)\.(\d+)\.(\d+)$")
+
+
+class ReleaseNotes(SphinxDirective):
+    required_arguments: ClassVar = 1
+
+    def run(self) -> Sequence[nodes.Node]:
+        dir_ = Path(self.arguments[0])
+        # resolve relative dir
+        if not dir_.is_absolute():
+            src_file = Path(self.get_source_info()[0])
+            if not src_file.is_file():
+                msg = f"Cannot find relative path to: {src_file}"
+                raise self.error(msg)
+            dir_ = src_file.parent / self.arguments[0]
+        if not dir_.is_dir():
+            msg = f"Not a directory: {dir_}"
+            raise self.error(msg)
+
+        versions = sorted(
+            (
+                (Version(f.stem), f)
+                for f in dir_.iterdir()
+                if FULL_VERSION_RE.match(f.stem)
+            ),
+            reverse=True,  # descending
+        )
+        version_groups = itertools.groupby(
+            versions, key=lambda vf: (vf[0].major, vf[0].minor)
+        )
+        for (major, minor), versions in version_groups:
+            self.render_version_group(major, minor, versions)
+        return []
+
+    def render_version_group(
+        self, major: int, minor: int, versions: Iterable[tuple[Version, Path]]
+    ) -> None:
+        target = nodes.target(
+            ids=[f"v{major}-{minor}"],
+            names=[f"v{major}.{minor}"],
```

Comment:
I was/have always been wondering about this - why use `v` some places and not in others?

Reply:
i.e., it is not used here https://github.com/scverse/anndata/pull/1604/files#diff-bdbc86648710a391d76972ef99a2b1cb8587ec09d9b4aea5d403497f5cf2d806R63-R65

Reply:
For me, I find it clearer to only do one or the other (`v` or no `v`)

Reply:
Aside from the two refs to this, nothing else uses these and we use `v` nowhere else to refer to releases from what I can tell

Reply:
>why use v some places and not in others

MyST didn’t turn the `(x.y)=` (_target_) syntax into a URL anchor unless you added the `v`.

So Isaac spelled the `x.y` headings “Version x.y” which results in `…/release-notes#version-x-y` slugs.
I didn’t want the word “version” on the page hundreds of times, so I experimented until I figured out that MyST *will* respect the target syntax if there’s at least one letter.

The changes in this PR are to stay consistent with the slugs we already have because that’s the whole point: that links to to `…/release-notes#version-0.10` or `…/release-notes#v0.10.8` are stable.

Reply:
> MyST doesn’t turn the (x.y)= (target) syntax into a URL anchor unless you add letters.

I didn't know you need the letters, ok.

That makes sense, I didn't know the history of this.